book/src/out-of-order-blocks.md

```diff
@@ -0,0 +1,125 @@
+# Out Of Order Blocks
+
+The [deterministic leader schedule](../tree/master/book/src/leader-rotation.md)
+creates a fixed schedule for block production.  The main problems with this
+approach are as follows:
+
+* The leader schedule has to balance both performance and liveness requirements.
+The epoch needs to be long enough to guarantee liveness.
+
+* The cluster has to wait for unavailable nodes.
+
+* The cluster's liveness depends on the availability of scheduled nodes.
+
+* Attackers know in advance all of the leaders that propose a block.
+
+The general problems of DPOS — that a cluster requires an interactive protocol
+for account registration, delegation, and voting that a set of leaders can
+censor — is not addressed in this design.
+
+## Random Leaders
+
+Leaders can generate a *golden ticket* to propose a block.
```

Comment:
Is it the leader for the current slot? Or the node that's trying to be the leader for the proposed block? I am guessing its former, just want to be sure.

Reply:
@pgarg66 yea, the former.